addressing/README.md

```diff
@@ -147,6 +147,33 @@ within it. Using our example above, decapsulating either `/tcp/1234/ws` _or_
 unsurprising if you consider the utility of the `/ip4/7.7.7.7/ws` address that
 would result from simply removing the `tcp` component.
 
+### The security multiaddr
+
+Peers MAY advertise their addresses without the security protocol, e.g.
+`/ip4/6.6.6.6/tcp/1234/` or `/ip4/6.6.6.6/udp/1234/quic`. The security handshake
+protocol is then negotiated using the [multistream](../multistream.md). This is
+the way the libp2p handshake worked until mid 2021.
+This poses a security problem, as the negotiation was not authenticated and
+therefore susceptible to man-in-the-middle attacks. A MITM could modify the list
+of supported handshake protocols, thereby forcing a downgrade to a (potentially)
+less secure handshake protocol. Note that since QUIC is standardized to use
+TLS 1.3, no handshake protocol needs to be negotiated when using QUIC.
+
+Peers SHOULD encapsulate the security protocol in the addresses they advertise,
+for example `/ip4/6.6.6.6/tcp/1234/tls` for a TLS 1.3 server listening on TCP
+port 1234 and `/ip4/6.6.6.6/tcp/1235/noise` for a Noise server listening on TCP
+port 1235. QUIC multiaddrs remain unchanged.
+The nodes jump straight into a cryptographic handshake, thus curtailing the
+possibility of packet-inspection-based censorship and dynamic downgrade attacks.
+This also applies to circuit addresses: the security protocol is encoded in the
+`<destination address>` as defined in [`p2p-circuit` Relay Addresses](#p2p-circuit-relay-addresses).
+
+Implementations using [Protocol Select](https://github.com/libp2p/specs/pull/349/)
+(**TODO**: update link) MUST encapsulate the security protocol in the multiaddr.
+Note that it’s valid to assume that any node that encapsulated the security
+protocol in their multiaddr also supports Protocol Select.
```

Comment:
I thought last consensus was on not tying the two together. Again, I am fine either way, I only want to be consistent.

https://github.com/libp2p/specs/pull/349 currently includes:

> Note: A peer MAY advertise a Multiaddr that includes a secure channel handshake
protocol like `/noise` even if it doesn't support Protocol Select. See
[Heuristic section](#heuristic) below for details on how listeners can
differentiate the negotiation protocol spoken by the dialer on incoming
connections.

Reply:
The intention is to not tie them together, in the sense that advertising a security-enabled multiaddr doesn't mean that you speak Protocol Select. You can advertise a security-enabled multiaddr and still speak multistream.

However, I thought we agreed to require you to advertise a security-enabled multiaddr if you want to speak Protocol Select. Maybe that's not necessary though and you could use the Protocol Select protobufs to negotiate your handshake protocol?

Reply:
### Protocol Select requiring secure multiaddr component

> However, I thought we agreed to require you to advertise a security-enabled multiaddr if you want to speak Protocol Select.

Yes. We agreed on that.

>  Maybe that's not necessary though and you could use the Protocol Select protobufs to negotiate your handshake protocol?

I don't see a benefit of supporting this. Anything I am missing?

### Secure multiaddr component implying Protocol Select

> in the sense that advertising a security-enabled multiaddr doesn't mean that you speak Protocol Select.

> Note that it’s valid to assume that any node that encapsulated the security protocol in their multiaddr also supports Protocol Select.

Are these two sentences not in direct conflict?

Reply:
### Protocol Select requiring secure multiaddr component

> However, I thought we agreed to require you to advertise a security-enabled multiaddr if you want to speak Protocol Select.

Yes. We agreed on that.

>  Maybe that's not necessary though and you could use the Protocol Select protobufs to negotiate your handshake protocol?

I don't see a benefit of supporting this. Anything I am missing?

### Secure multiaddr component implying Protocol Select

> in the sense that advertising a security-enabled multiaddr doesn't mean that you speak Protocol Select.

> Note that it’s valid to assume that any node that encapsulated the security protocol in their multiaddr also supports Protocol Select.

Are these two sentences not in direct conflict?

Reply:
There was a "not" missing... Fixed.